### DISCUSSION TITLE
meson watch

### DISCUSSION THREAD

**Discussion by Freed-Wu** [mesonbuild/meson#13143] [Q&A]
Is it possible to have a `meson watch -Cbuild_dir`, when source_dir is changed, it will can `meson complie -Cbuild_dir`?

**Comment by eli-schwartz** [comment]
On Linux you can do this with inotifywait.

It is unlikely to be added to meson as a `meson watch` command, if only because inotify is neither portable to Windows nor particularly usable from python without third-party dependencies. And meson is not adding a dependency on third-party python libraries.

**Reply by Freed-Wu** [reply]
I know typescript's `tsc` have `tsc -w` (`tsc --watch`) and it can work for all platforms. Is there any different way to support all platforms?

**Reply by eli-schwartz** [reply]
Typescript is the compiler itself, not the build system generator, so that would be more comparable to `ninja` probably.

Which is actually an interesting point because ninja actually knows which files it is interested in and could subscribe to an inotify watch on those files. Meson doesn't know the list of build inputs anymore -- it knew during configuration but loses that information as soon as `meson setup` ends.

**Reply by Freed-Wu** [reply]
So it should be realized in `ninja`, as `ninja -w target_name`, then meson can use it?

**Reply by eli-schwartz** [reply]
I just checked and there's already a feature request for ninja for this:

https://github.com/ninja-build/ninja/issues/833

Open since 2014, but it did recently get added to a milestone, perhaps that means they are interested in the feature now?
